### PR TITLE
feat: add --seed CLI flag for deterministic generation of dynamic examples

### PR DESCRIPTION
**Pull request by ilanashapiro**
**Summary**

Addresses issue #2590

We introduce a new CLI flag --seed that the user sets to a string. This string is feed as a seed into json-schema-faker as described in https://github.com/json-schema-faker/json-schema-faker/issues/111, and allows for reproducible dynamic examples, which is useful in testing. The user can also set the seed in the Prefer header.

NOTE: if you are using --ignoreExamples for Prism to statically generate examples conforming to a schema, you do NOT need to use the --seed flag as this will have no effect. --ignoreExamples uses json-schema-sampler to statically generate examples, not json-schema-faker:

https://github.com/stoplightio/prism/blob/0d12d25b1907385a6f84a3634c8aedd56f113ea4/packages/http/src/mocker/generator/JSONSchema.ts#L107

and json-schema-sampler is **already deterministic**: https://github.com/stoplightio/json-schema-sampler#:~:text=Deterministic%20(given%20a%20particular%20input%2C%20will%20always%20produce%20the%20same%20output).

**Checklist**

- The basics
  - [x] I tested these changes manually in my local or dev environment
- Tests
  - [x] Added or updated
  - [ ] N/A
- Event Tracking
  - [ ] I added event tracking and followed the event tracking guidelines
  - [x] N/A
- Error Reporting
  - [ ] I reported errors and followed the error reporting guidelines
  - [x] N/A

**Screenshots**
We mock the following .yaml spec:
```
openapi: 3.0.2
paths:
  /pets:
    get:
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                '$ref': '#/schemas/Pet'
              examples:
                cat:
                  summary: An example of a cat
                  value:
                    id: "string_id"
                    name: 123
                    photoUrls: [[]]
                    status: 135

schemas:
  Pet:
      type: object
      required:
      - name
      - photoUrls
      properties:
        id:
          type: integer
          format: int64
        name:
          type: string
          example: doggie
        photoUrls:
          type: array
          xml:
            name: photoUrl
            wrapped: true
          items:
            type: string
        status:
          type: string
          description: pet status in the store
          enum:
          - available
          - pending
          - sold
```

Mocking with --dynamic and --seed gives us reproducible dynamic responses:
image
image

Vs mocking with just --dynamic gives us random/variable dynamic responses:
image
image

Finally, if you mock in dynamic mode but without a seed specified, you can still add the seed in the Prefer header. Note how in the following example, the first curl request doesn't add a Prefer header, but the second and third do and specify the same seed, and the second and third responses are identical to one another.
image

image

### CODE REVIEW

**Comment by ilanashapiro** [comment]
@dgellow

**Comment by ilanashapiro** [comment]
Hi @brendarearden! Could we request a review on this from someone on your team? Thank you!

**Review by dgellow — APPROVED** [review]
LGTM, thanks a lot Ilana 👍

**Comment by dgellow** [comment]
@lukaszzazulak Can we do something to help speed up a review of this PR?

**Comment by ilanashapiro** [comment]
Following up -- could @lukaszm-sb or @lukaszzazulak please help us get a review on this either by you or someone else on the Prism team? Thanks!

**Comment by tesanchez** [comment]
@dgellow Thanks for your commitment here. We are looking into some issues we have with the automation process for deployment for some of the dependencies. As soon as we have some free time we will try to get to reviewing this PR.

**Comment by jack-bliss** [comment]
this is awesome, thanks!

**Review by lukaszzazulak — APPROVED** [review]

**Comment by lukaszzazulak** [comment]
@ilanashapiro sorry for the delay, could you resolve conflicts ?

**Comment by ilanashapiro** [comment]
@lukaszzazulak conflicts are resolved! thank you for approving the changes!

File: yarn.lock [hunk 1]
```diff
@@ -3153,9 +3153,9 @@ fast-safe-stringify@^2.0.8, fast-safe-stringify@^2.1.1:
   integrity sha512-W+KJc2dmILlPplD/H4K9l9LcAHAfPtP6BY84uVLXQ6Evcz9Lcg33Y2z1IVblT6xdY54PXYVHEv+0Wpq8Io6zkA==
 
 fast-xml-parser@^4.2.0:
-  version "4.4.1"
-  resolved "https://registry.yarnpkg.com/fast-xml-parser/-/fast-xml-parser-4.4.1.tgz#86dbf3f18edf8739326447bcaac31b4ae7f6514f"
-  integrity sha512-xkjOecfnKGkSsOwtZ5Pz7Us/T6mrbPQrq0nh+aCO5V9nk5NLWmasAHumTKjiPJPWANe+kAZ84Jc8ooJkzZ88Sw==
+  version "4.3.2"
```
Comment (lukaszzazulak) [inline yarn.lock:3156]: is has lately been upgraded to 4.4.1 (https://github.com/stoplightio/prism/pull/2570), something went wrong with the merge ?

**Review by lukaszzazulak — CHANGES_REQUESTED** [review]

File: yarn.lock (hunk 1 above)
Comment (ilanashapiro) [reply yarn.lock]: @lukaszzazulak  Hmm sorry unsure why this happened, I pulled the changes from the master branch and did a fresh reinstall of all packages based on the updated package.json file I received. I just copy-pasted the code from #2570 into the yarn.lock file and hopefully this resolves things!

**Review by lukaszzazulak — APPROVED** [review]

**Comment by dgellow** [comment]
Ilana's PR has been merged 🎉. It will be part of the next release: https://github.com/stoplightio/prism/pull/2606.

**Pull request merged**